### PR TITLE
Use http_client_factory for guzzle requests instead of GuzzleClient

### PR DESCRIPTION
**Pull request by snwalkunde**
- https://www.drupal.org/project/eloqua_api_redux/issues/3047348

### CODE REVIEW

File: src/Service/EloquaApiClient.php [hunk 1]
```diff
@@ -54,14 +54,18 @@ class EloquaApiClient {
...
    * @param \Drupal\Core\Cache\CacheBackendInterface $cacheBackend
    *   Cache Backend.
+   * @param \Drupal\Core\Http\ClientFactory $http_client_factory
+   *   A Guzzle client object.
    */
   public function __construct(ConfigFactory $config,
                               LoggerChannelFactoryInterface $loggerFactory,
-                              CacheBackendInterface $cacheBackend) {
+                              CacheBackendInterface $cacheBackend,
+                              ClientFactory $http_client_factory) {
```
Comment (dakkusingh) [inline src/Service/EloquaApiClient.php]: might want to try to use camel case vars to keep things consistent :)

File: src/Service/EloquaApiClient.php [hunk 2]
```diff
@@ -152,7 +156,7 @@ public function getAccessTokenByRefreshToken() {
    */
   private function doTokenRequest(array $params, $res = 'token') {
     // Guzzle Client.
-    $guzzleClient = new GuzzleClient([
+    $guzzleClient = $this->httpClientFactory->fromOptions([
```
Comment (dakkusingh) [inline src/Service/EloquaApiClient.php:159]: surely this shouldnt be `$guzzleClient` as per the line 46 where its `protected $httpClientFactory;`

File: src/Service/EloquaApiClient.php [hunk 3]
```diff
@@ -270,7 +274,7 @@ public function getBaseUrl() {
   private function doBaseUrlRequest() {
     // TODO Ideally merge all the Guzzle requests into one generic method.
     // Guzzle Client.
-    $guzzleClient = new GuzzleClient([
+    $guzzleClient = $this->httpClientFactory->fromOptions([
```
Comment (dakkusingh) [inline src/Service/EloquaApiClient.php:277]: surely this shouldnt be `$guzzleClient` as per the line 46 where its `protected $httpClientFactory;`

File: src/Service/EloquaApiClient.php [hunk 4]
```diff
@@ -367,7 +371,7 @@ private function getEloquaApiCache($type) {
   public function doEloquaApiRequest($verb, $endpoint, $body = NULL, $queryParams = NULL) {
     // TODO Ideally merge all the Guzzle requests into one generic method.
     // Guzzle Client.
-    $guzzleClient = new GuzzleClient([
+    $guzzleClient = $this->httpClientFactory->fromOptions([
```
Comment (dakkusingh) [inline src/Service/EloquaApiClient.php:374]: surely this shouldnt be `$guzzleClient` as per the line 46 where its `protected $httpClientFactory;`

File: src/Service/EloquaApiClient.php (hunk 1 above)
Comment (snwalkunde) [reply src/Service/EloquaApiClient.php]: Updated

**Review by dakkusingh — APPROVED** [review]
Looks good :)

**Pull request merged**